### ISSUE TITLE
Large file problem

### ISSUE THREAD

**Issue by 0ldi** [Franiac/TwitchLeecher#208]
I tried to download the 47:59:54 stream, but after the completion (no errors) the video duration was 26:30:59 with the size of 104 Gb
Here is my question, is there any limitation on the size / duration?
this video: https://www.twitch.tv/videos/209692291
I tried 2 times, and the files were identical
-last .NET Framework
-64 Bit Windows 10
-64 Bit TwitchLeecher version v1.4.2

**Comment by Franiac** [comment]
There is no limitation but FFMPEG is not very error-prone. Twitch streams are not the best source when it comes to error-free static streams. A single faulted frame in the stream and your download is wasted, that's how it is unfortunately. A web player can easily throw the frame away and go on to the next one but FFMPEG is not capable of ignoring errors as far as I know (I am NOT a FFMPEG expert). I strongly recommend you to use the cropping feature and split the VOD into multiple smaller portions (like 4-5 hours). Such long streams are quite uncommon but if there are any the are always problematic with TL, I know that. Everytime Games Done Qick is over I get a lot of issues ;)

**Comment by Franiac** [comment]
Oh and you know the stream is faulted at 26:30:59.... that's where FFMPEG decides to leave the game... so just crop that 10s part out of there...

**Comment by 0ldi** [comment]
The problem is that this is not my channel (and why do I need TL to download my record, if it can be downloaded via the website of the twitch), so I can not divide it into clips.
But exactly this problem, I decided due https://github.com/MrS0m30n3/youtube-dl-gui
The downloaded video is the same size as with TL, but has a correct duration. BUT, if this file is opened my main PotPlayer, after 26:30:59 the image freezes, BUT if opened with MPC, all video is played normally. This of course confirms the fact that the file is broken. 
I hope with crop there will be no problems. And nevertheless, if another APP has coped, then TL has some problem.

**Comment by Franiac** [comment]
@0ldi I am not sure if I understand you correctly. It does not matter what channel it is, the cropping feature in TL is always available and works on all VODs, even if they are not on your own channel.

TL never has a problem if such an error occures. TL uses FFMPEG, youtube-dl uses FFEMPEG, every other VOD downloader uses FFMPEG. It is a question about how you configure FFMPEG and run it in your app. I am a C# software engineer not a FFMPEG pro so I can only do what I know. If there is a problem with video processing, I need someone who knows FFMPEG and all the parameters you can configure it with.
In other words: If youtube-dl manages it to download the video flawlessly, it just uses a different FFMPEG configuration than TL. If we can find out what that configuration is, I can make TL work the same way.

Just keep in mind that whatever app you are using for video download, they 99% of the time are pure FFMPEG wrappers with a more or less fancy GUI. TL is just a more fancy one :D
